tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -540,18 +552,51 @@ shared_ptr<CKKSTensor> CKKSTensor::_dot_inplace(
     }
 }
 
-shared_ptr<CKKSTensor> CKKSTensor::dot_inplace(
-    const shared_ptr<CKKSTensor>& other) {
-    auto other_shape = other->shape();
-
-    return this->_dot_inplace(other, other_shape);
-}
-
 shared_ptr<CKKSTensor> CKKSTensor::dot_plain_inplace(
     const PlainTensor<double>& other) {
+    auto this_shape = this->shape();
     auto other_shape = other.shape();
-
-    return this->_dot_inplace(other, other_shape);
+    if (this_shape.size() == 1) {
+        if (other_shape.size() == 1) {  // 1D-1D
+            // inner product
+            this->_mul_inplace(other);
+            this->sum_inplace();
+            return shared_from_this();
+        } else if (other_shape.size() == 2) {  // 1D-2D
+            if (this_shape[0] != other_shape[0])
+                throw invalid_argument("can't perform dot: dimension mismatch");
+            // TODO: remove boradcast when implemented in _mul
+            this->_data.reshape_inplace(vector<size_t>({this_shape[0], 1}));
+            this->_data.broadcast_inplace(other_shape);
+            this->_mul_inplace(other);
+            this->sum_inplace();
+            return shared_from_this();
+        } else {
+            throw invalid_argument(
+                "don't support dot operations of more than 2 dimensions");
+        }
+    } else if (this_shape.size() == 2) {
+        if (other_shape.size() == 1) {  // 2D-1D
+            if (this_shape[1] != other_shape[0])
+                throw invalid_argument("can't perform dot: dimension mismatch");
+            // TODO: remove boradcast when implemented in _mul
+            auto other_copy = other;
```

Comment:
maybe
```
auto other_copy = other.reshape(...
```
directly?

Reply:
other here is a const and can't call reshape directly, just kept this, but the rest is updated